### ISSUE TITLE
ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider fails and gets disabled

### ISSUE THREAD

**Issue by rodrmoya** [mono/monodevelop#5765]
Not sure how to fire it, but it's been happening several times today:

<img width="1100" alt="screen shot 2018-08-24 at 13 29 46" src="https://user-images.githubusercontent.com/1873039/44582388-cca8fc00-a7a1-11e8-910f-6aea4ca6e308.png">

This is on latest 7.7 from beta channel

> VS bug [#670909](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/670909)

**Comment by Therzok** [comment]
Stacktrace from the logs please.

**Comment by rodrmoya** [comment]
```ERROR [2018-08-24 13:24:20Z]: Roslyn error: StorageDatabase_Exceptions Kind=Microsoft.CodeAnalysis.SQLite.Interop.SqlException|Reason=Microsoft.CodeAnalysis.SQLite.Interop.SqlException: database table is locked: SolutionData1
database table is locked
  at Microsoft.CodeAnalysis.SQLite.Interop.SqlConnection.Throw (SQLitePCL.sqlite3 handle, Microsoft.CodeAnalysis.SQLite.Interop.Result result) [0x00000] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlConnection.cs:284 
  at Microsoft.CodeAnalysis.SQLite.Interop.SqlConnection.ThrowIfNotOk (SQLitePCL.sqlite3 handle, Microsoft.CodeAnalysis.SQLite.Interop.Result result) [0x00003] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlConnection.cs:276 
  at Microsoft.CodeAnalysis.SQLite.Interop.SqlConnection.ThrowIfNotOk (Microsoft.CodeAnalysis.SQLite.Interop.Result result) [0x00000] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlConnection.cs:270 
  at Microsoft.CodeAnalysis.SQLite.Interop.SqlConnection.ThrowIfNotOk (System.Int32 result) [0x00000] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlConnection.cs:267 
  at Microsoft.CodeAnalysis.SQLite.Interop.SqlStatement.Reset () [0x00000] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlStatement.cs:43 
  at Microsoft.CodeAnalysis.SQLite.Interop.ResettableSqlStatement.Dispose () [0x00000] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/ResettableSqlStatement.cs:34 
  at Microsoft.CodeAnalysis.SQLite.SQLitePersistentStorage+Accessor`3[TKey,TWriteQueueKey,TDatabaseId].InsertOrReplaceBlob (Microsoft.CodeAnalysis.SQLite.Interop.SqlConnection conection, TDatabaseId dataId, System.Byte[] bytes, System.Int32 length) [0x00046] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage.Accessor.cs:228 
  at Microsoft.CodeAnalysis.SQLite.SQLitePersistentStorage+Accessor`3+<>c__DisplayClass12_1[TKey,TWriteQueueKey,TDatabaseId].<WriteStreamAsync>b__0 (Microsoft.CodeAnalysis.SQLite.Interop.SqlConnection con) [0x00000] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage.Accessor.cs:120 
  at Microsoft.CodeAnalysis.SQLite.SQLitePersistentStorage+<>c__DisplayClass74_0.<ProcessWriteQueue>b__0 () [0x00015] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_WriteBatching.cs:229 
  at Microsoft.CodeAnalysis.SQLite.Interop.SqlConnection.RunInTransaction (System.Action action) [0x00086] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlConnection.cs:168 
  at Microsoft.CodeAnalysis.SQLite.SQLitePersistentStorage.ProcessWriteQueue (Microsoft.CodeAnalysis.SQLite.Interop.SqlConnection connection, Microsoft.CodeAnalysis.PooledObjects.ArrayBuilder`1[T] writesToProcess) [0x00039] in /_/src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_WriteBatching.cs:225 
ERROR [2018-08-24 13:26:28Z]: Roslyn error: Extension_Exception ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider : Object reference not set to an instance of an object
  at RefactoringEssentials.CSharp.CodeRefactorings.ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider+<ComputeRefactoringsAsync>d__0.MoveNext () [0x003c4] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider.cs:63 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:447 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService+<GetRefactoringFromProviderAsync>d__8.MoveNext () [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
ERROR [2018-08-24 13:26:38Z]: Roslyn error: Extension_Exception ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider : Object reference not set to an instance of an object
  at RefactoringEssentials.CSharp.CodeRefactorings.ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider+<ComputeRefactoringsAsync>d__0.MoveNext () [0x003c4] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider.cs:63 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:447 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService+<GetRefactoringFromProviderAsync>d__8.MoveNext () [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
ERROR [2018-08-24 13:27:11Z]: Roslyn error: Extension_Exception ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider : Object reference not set to an instance of an object
  at RefactoringEssentials.CSharp.CodeRefactorings.ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider+<ComputeRefactoringsAsync>d__0.MoveNext () [0x003c4] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider.cs:63 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:447 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService+<GetRefactoringFromProviderAsync>d__8.MoveNext () [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
INFO [2018-08-24 13:27:22Z]: ANDROID: Found Xamarin.Android 9.0.0.17
ERROR [2018-08-24 13:38:09Z]: An unhandled exception has occurred. Terminating Visual Studio? False
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.CSharp.Completion.CSharpCompletionTextEditorExtension.AddImportCompletionData (Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery.CSharpSyntaxContext ctx, MonoDevelop.Ide.CodeCompletion.CompletionDataList result, Microsoft.CodeAnalysis.SemanticModel semanticModel, System.Int32 position, System.Threading.CancellationToken cancellationToken) [0x000ab] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CSharpCompletionTextEditorExtension.cs:283 
  at MonoDevelop.CSharp.Completion.CSharpCompletionTextEditorExtension+<InternalHandleCodeCompletion>d__36.MoveNext () [0x0060e] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CSharpCompletionTextEditorExtension.cs:470 
ERROR [2018-08-24 13:39:31Z]: An unhandled exception has occurred. Terminating Visual Studio? False
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.CSharp.Completion.CSharpCompletionTextEditorExtension.AddImportCompletionData (Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery.CSharpSyntaxContext ctx, MonoDevelop.Ide.CodeCompletion.CompletionDataList result, Microsoft.CodeAnalysis.SemanticModel semanticModel, System.Int32 position, System.Threading.CancellationToken cancellationToken) [0x000ab] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CSharpCompletionTextEditorExtension.cs:283 
  at MonoDevelop.CSharp.Completion.CSharpCompletionTextEditorExtension+<InternalHandleCodeCompletion>d__36.MoveNext () [0x0060e] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CSharpCompletionTextEditorExtension.cs:470 
ERROR [2018-08-24 13:40:38Z]: Exception in RegexCompletionProvider
System.Threading.Tasks.TaskCanceledException: A task was canceled.
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00020] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:179 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:535 
  at Microsoft.CodeAnalysis.SemanticModelWorkspaceService.SemanticModelWorkspaceServiceFactory+SemanticModelService+<GetSemanticModelForNodeAsync>d__3.MoveNext () [0x00211] in /_/src/Workspaces/Core/Portable/SemanticModelWorkspaceService/SemanticModelWorkspaceServiceFactory.cs:58 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0001a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:176 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:535 
  at Microsoft.CodeAnalysis.Shared.Extensions.DocumentExtensions+<GetSemanticModelForSpanAsync>d__2.MoveNext () [0x00267] in /_/src/Workspaces/Core/Portable/Shared/Extensions/DocumentExtensions.cs:62 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0001a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:176 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:535 
  at MonoDevelop.CSharp.Completion.Provider.RegexCompletionProvider+<ProvideCompletionsAsync>d__4.MoveNext () [0x00078] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/RegexCompletionProvider.cs:78 
ERROR [2018-08-24 13:40:38Z]: Exception in RegexCompletionProvider
System.Threading.Tasks.TaskCanceledException: A task was canceled.
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00020] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:179 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:535 
  at Microsoft.CodeAnalysis.SemanticModelWorkspaceService.SemanticModelWorkspaceServiceFactory+SemanticModelService+<GetSemanticModelForNodeAsync>d__3.MoveNext () [0x00211] in /_/src/Workspaces/Core/Portable/SemanticModelWorkspaceService/SemanticModelWorkspaceServiceFactory.cs:58 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0001a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:176 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:535 
  at Microsoft.CodeAnalysis.Shared.Extensions.DocumentExtensions+<GetSemanticModelForSpanAsync>d__2.MoveNext () [0x00267] in /_/src/Workspaces/Core/Portable/Shared/Extensions/DocumentExtensions.cs:62 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0001a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:176 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:535 
  at MonoDevelop.CSharp.Completion.Provider.RegexCompletionProvider+<ProvideCompletionsAsync>d__4.MoveNext () [0x00078] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/RegexCompletionProvider.cs:78 
ERROR [2018-08-24 13:49:56Z]: Async operation failed
System.AggregateException: One or more errors occurred. ---> System.ArgumentException: Metadata is not referenced.
  at Microsoft.CodeAnalysis.Workspace.CheckProjectHasMetadataReference (Microsoft.CodeAnalysis.ProjectId projectId, Microsoft.CodeAnalysis.MetadataReference metadataReference) [0x00024] in /_/src/Workspaces/Core/Portable/Workspace/Workspace.cs:1764 
  at Microsoft.CodeAnalysis.Workspace.OnMetadataReferenceRemoved (Microsoft.CodeAnalysis.ProjectId projectId, Microsoft.CodeAnalysis.MetadataReference metadataReference) [0x0001c] in /_/src/Workspaces/Core/Portable/Workspace/Workspace.cs:624 
  at MonoDevelop.Ide.TypeSystem.MonoDevelopWorkspace+ProjectData.OnMetadataReferenceUpdated (System.Object sender, System.EventArgs args) [0x00031] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopWorkspace.cs:542 
  at (wrapper delegate-invoke) <Module>.invoke_void_object_EventArgs(object,System.EventArgs)
  at MonoDevelop.Ide.TypeSystem.MonoDevelopMetadataReference.OnUpdatedOnDisk (System.Object sender, MonoDevelop.Core.FileEventArgs e) [0x00024] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopMetadataReference.cs:55 
  at (wrapper delegate-invoke) System.EventHandler`1[MonoDevelop.Core.FileEventArgs].invoke_void_object_TEventArgs(object,MonoDevelop.Core.FileEventArgs)
  at MonoDevelop.Core.EventQueue+EventData`1[TArgs].Invoke () [0x00000] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/FileService.cs:860 
  at MonoDevelop.Core.EventQueue+<>c__DisplayClass9_0.<Thaw>b__0 () [0x00050] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/FileService.cs:933 
  at MonoDevelop.Core.Runtime.RunInMainThread (System.Action action) [0x0000d] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs:335 
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.ArgumentException: Metadata is not referenced.
  at Microsoft.CodeAnalysis.Workspace.CheckProjectHasMetadataReference (Microsoft.CodeAnalysis.ProjectId projectId, Microsoft.CodeAnalysis.MetadataReference metadataReference) [0x00024] in /_/src/Workspaces/Core/Portable/Workspace/Workspace.cs:1764 
  at Microsoft.CodeAnalysis.Workspace.OnMetadataReferenceRemoved (Microsoft.CodeAnalysis.ProjectId projectId, Microsoft.CodeAnalysis.MetadataReference metadataReference) [0x0001c] in /_/src/Workspaces/Core/Portable/Workspace/Workspace.cs:624 
  at MonoDevelop.Ide.TypeSystem.MonoDevelopWorkspace+ProjectData.OnMetadataReferenceUpdated (System.Object sender, System.EventArgs args) [0x00031] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopWorkspace.cs:542 
  at (wrapper delegate-invoke) <Module>.invoke_void_object_EventArgs(object,System.EventArgs)
  at MonoDevelop.Ide.TypeSystem.MonoDevelopMetadataReference.OnUpdatedOnDisk (System.Object sender, MonoDevelop.Core.FileEventArgs e) [0x00024] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopMetadataReference.cs:55 
  at (wrapper delegate-invoke) System.EventHandler`1[MonoDevelop.Core.FileEventArgs].invoke_void_object_TEventArgs(object,MonoDevelop.Core.FileEventArgs)
  at MonoDevelop.Core.EventQueue+EventData`1[TArgs].Invoke () [0x00000] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/FileService.cs:860 
  at MonoDevelop.Core.EventQueue+<>c__DisplayClass9_0.<Thaw>b__0 () [0x00050] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/FileService.cs:933 
  at MonoDevelop.Core.Runtime.RunInMainThread (System.Action action) [0x0000d] in /Users/vsts/agent/2.138.4/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs:335 <---
```

**Comment by mrward** [comment]
Just had this today with - do not have a repro - was making changes to FileService in monodevelop:

VS Mac 7.7.0.1624
Git revision: 3c69dfbf2f01b70820efbfa9d113166cf18d0713
Build date: 2018-10-12 19:44:04+00
Build branch: release-7.7
Xamarin extensions: 9385f42517cbb100c3c8124faac82dadeb9a79a0

```
Extension_Exception ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider : Object reference not set to an instance of an object
  at RefactoringEssentials.CSharp.CodeRefactorings.ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider+<ComputeRefactoringsAsync>d__0.MoveNext () [0x003c4] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/ToAbstractVirtualNonVirtualConversionCodeRefactoringProvider.cs:63 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService+<GetRefactoringFromProviderAsync>d__8.MoveNext () [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
```

**Comment by mrward** [comment]
OK I have a repro. In FileService.cs if you click inside the string 'FileServiceErrorHandler' at the bottom of the file that seems to trigger the error.

    public delegate bool FileServiceErrorHandler (string message, Exception ex);

You can reproduce the same error in a console app. Copy the above line into Program.cs outside the MainClass class.

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.840 (master)

Pull Request #6317 merged by: Lluis Sanchez
Author: mono
Commit: 617c62128a87dd128697031315d279b55d38aaf5 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1633 (release-7.7)

Pull Request #6318 merged by: Lluis Sanchez
Author: mono
Commit: 3f9ef1be4081a9e0bf3319a5a6a8d9aee2803014 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.471 (release-8.0)

Pull Request #6317 merged by: Lluis Sanchez
Author: mono
Commit: 617c62128a87dd128697031315d279b55d38aaf5 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #6317 merged by: Lluis Sanchez
Author: mono
Commit: 617c62128a87dd128697031315d279b55d38aaf5 (xamarin/monodevelop)
